dev/OAuth/AuthRequestAsyncOperation.cpp

```diff
@@ -0,0 +1,574 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include <pch.h>
+#include "common.h"
+
+#include "OAuth2Manager.h"
+#include "AuthRequestAsyncOperation.h"
+#include "AuthRequestResult.h"
+
+using namespace std::literals;
+using namespace winrt::Microsoft::Security::Authentication::OAuth;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Security::Cryptography;
+
+AuthRequestAsyncOperation::AuthRequestAsyncOperation(const winrt::hstring& state)
+{
+    try
+    {
+        if (state.empty())
+        {
+            while (true)
```

Comment:
Why while loop is written here?

Reply:
Updated the code to have state inside the loop. The loop is here to ensure we get a unique state value.